client_cli/pytests/README.md

```diff
@@ -17,21 +17,42 @@ The Iroha 2 Test Model consists of several test categories that cover different
 
 - **Atomicity:** Test cases for transaction atomicity, including multiple instructions within a single transaction, paired instructions, and invalid instructions.
 
-## How to use
-At first, you need to installed and running [Iroha 2](https://hyperledger.github.io/iroha-2-docs/guide/install.html), and also need to have built [Client CLI](https://hyperledger.github.io/iroha-2-docs/guide/build.html)
+## Usage
 
-## Configuration
+1. Setup test environment using [`test_env.py`](../../scripts/test_env.py):
```

Comment:
I think we can write a subsection about it, called "Custom testing environment". It might say that if `test_env.py` is not convenient, you can always set up Iroha network on your own, e.g. using docker compose configuration.

What do you think?

Reply:
> I think we can write a subsection about it, called "Custom testing environment". It might say that if `test_env.py` is not convenient, you can always set up Iroha network on your own, e.g. using docker compose configuration.
> 
> What do you think?

Agreed

Reply:
With `test_env.py` being the "blessed" option in this section, it would be nice to also include the configuration needed to make it work with it (or maybe make the pytests configuration integrate more seamlessly with it).

Right now to understand what command should you run to make pytests run against the iroha deployed by `test_env.py`,  you have to navigate to the section about configuration and somehow figure out that you need to set the `CLIENT_CLI_DIR` environment variable to `../../test` to point it to the directory created by `test_env.py`.

This can be improved at least by specifically spelling out the necessary value of this option to use with `test_env.py`.

Running the tests without specifying the environment variable says that it can't find the `config.json` file in the `client_cli` directory, without even suggesting that it understands `CLIENT_CLI_DIR` variable to change it.

This can be improved by, obviously, giving a hint as to why the specific path is chosen and specifying the `CLIENT_CLI_DIR`, or, alternatively, just not accepting the lack of `CLIENT_CLI_DIR` env, making its use deliberate.

In an ideal world, it would also be nice to have a single script that handles both the concern of spinning up an iroha instance and running tests, so that it can be put into a pre-commit hook or just ran as a stand-alone tool without concerning about the configuration

Reply:
> In an ideal world, it would also be nice to have a single script that handles both the concern of spinning up an iroha instance and running tests

Strongly agree. In this case, `test_env.py` might be re-written as this single script, which will:

1. Setup the tests, install poetry packages
2. Setup the env for tests
3. Run tests
4. Cleanup the env

Currently it seem to not make sense that `test_env.py` is separated from the pytests.

Reply:
> > In an ideal world, it would also be nice to have a single script that handles both the concern of spinning up an iroha instance and running tests
> 
> Strongly agree. In this case, `test_env.py` might be re-written as this single script, which will:
> 
> 1. Setup the tests, install poetry packages
> 2. Setup the env for tests
> 3. Run tests
> 4. Cleanup the env
> 
> Currently it seem to not make sense that `test_env.py` is separated from the pytests.

I generally agree too, but one note for general understanding. `test_env.py` is  separated because I expect that only we are doing the environment via `test_env.py`. In real life iroha will be running on isolated stands on remote servers and necessarily require or local execute `test_env.py` as-is not correct. We only need iroha running anywhere and on anything. And on client machine, `iroha_client_cli` and `config.json` are enough. And it is incorrect to make tests dependent on `cargo build`. But yes, I think we can add something to `test_env.py` just to make our lives easier.

Reply:
Also:

- https://github.com/hyperledger/iroha/issues/4176